src/ledger/backend/ethereumOnChainQuerier.ts

```diff
@@ -24,31 +25,43 @@ export class EthereumOnChainQuerier implements OnChainQuerier {
 			erc721.queryFilter(inboundFilter),
 		]);
 
-		outbound.sort((x, y) => x.blockNumber - y.blockNumber);
-		inbound.sort((x, y) => x.blockNumber - y.blockNumber);
-
-		const tokenOwnedMemo: { [key: string]: boolean } = {};
-
-		while (outbound.length > 0 || inbound.length > 0) {
-			if (
-				inbound.length === 0 ||
-				(outbound.length > 0 &&
-					outbound[0].blockNumber < inbound[0].blockNumber)
-			) {
-				const nextToken = outbound[0].args.tokenId.toString();
-				tokenOwnedMemo[nextToken] = false;
-				outbound.shift();
-			} else {
-				const nextToken = inbound[0].args.tokenId.toString();
-				tokenOwnedMemo[nextToken] = true;
-				inbound.shift();
-			}
+		interface transferEvent
+			extends TypedEvent<
+				[string, string, BigNumber] & {
+					from: string;
+					to: string;
+					tokenId: BigNumber;
+				}
+			> {}
+
+		const compareTransactions = (
+			x: transferEvent,
+			y: transferEvent,
```

Comment:
If you do
```suggestion
			x: { blockNumber: number; transactionIndex: number },
			y: { blockNumber: number; transactionIndex: number },
```
, like I suggested, you can delete lines 28-25 (`interface transferEvent`) and cleanup some imports. Now that you use this function only once, you could even pass it directly to `ownershipHistory.sort(...)` and wouldn't need to specify the types of `x` and `y`.